plugins/woocommerce/client/blocks/assets/js/blocks/cart/test/editor-integration.tsx

```diff
@@ -0,0 +1,113 @@
+	beforeAll( async () => {
+		// Register a checkout filter to allow `core/table` block in all Cart inner blocks,
+		// add `core/audio` into the woocommerce/cart-order-summary-block specifically
+		await act( async () => {
+			registerCheckoutFilters( 'woo-test-namespace', {
+				// @ts-expect-error - The types for the checkout filters are not defined.
+				additionalCartCheckoutInnerBlockTypes: (
+					value: string[],
+					extensions,
+					{ block }: { block: string }
+				) => {
+					value.push( 'core/table' );
+					if ( block === 'woocommerce/cart-order-summary-block' ) {
+						value.push( 'core/audio' );
+					}
+					return value;
+				},
+			} );
+		} );
+	} );
+
+	it( 'inner blocks can be added/removed by filters', async () => {
+		await setup( {} );
+
+		// Verify Cart block is properly initialized in the editor.
+		expect( screen.getByLabelText( /^Block: Cart$/i ) ).toBeInTheDocument();
+
+		// Test Order Summary block - should have both Table and Audio options (specific filter applied).
+		await selectBlock( /^Block: Order Summary$/i );
+		const orderSummaryBlock = screen.getByLabelText(
+			/^Block: Order Summary$/i
+		);
+		const orderSummaryAddButton = getByLabelText(
+			orderSummaryBlock,
+			/^Add block$/i
+		);
+
+		// Open the block inserter for Order Summary.
+		await act( async () => {
+			await userEvent.click( orderSummaryAddButton );
+		} );
+
+		// Verify Table option is available (should be available on all blocks).
+		const tableOption = screen.getByRole( 'option', { name: /Table/i } );
+		expect( tableOption ).toBeInTheDocument();
+
+		// Verify Audio option is available (added only for order summary block).
+		const audioOption = screen.getByRole( 'option', { name: /Audio/i } );
+		expect( audioOption ).toBeInTheDocument();
+
+		// Test Filled Cart block - should only have Table option (no block-specific Audio filter).
+		const filledCartBlock = screen.getByLabelText( /Block: Filled Cart/i );
+		await act( async () => {
+			await userEvent.click( filledCartBlock );
+		} );
+
+		if ( ! filledCartBlock.parentElement ) {
+			throw new Error( 'Filled Cart block parent element not found.' );
+		}
+
+		// Find and click the first "Add block" button within the Filled Cart context.
+		const filledCartAddButtons = getAllByRole(
+			filledCartBlock.parentElement,
+			'button',
+			{
+				name: /^Add block$/i,
+			}
+		);
+
+		// Open the block inserter for Filled Cart.
+		await act( async () => {
+			await userEvent.click( filledCartAddButtons[ 0 ] );
+		} );
+
+		// Verify Table option is still available (general filter applies to all cart blocks).
+		const filledCartTableOption = screen.getByRole( 'option', {
+			name: /Table/i,
+		} );
+		expect( filledCartTableOption ).toBeInTheDocument();
+
+		// Verify Audio option is NOT available (block-specific filter only applies to Order Summary).
+		const filledCartAudioOption = screen.queryByRole( 'option', {
+			name: /Audio/i,
+		} );
```

Comment:
I am not getting the same gains here locally applying similar optimization as suggested in https://github.com/woocommerce/woocommerce/pull/58615/files#r2135663777